src/ctia/task/codegen.clj

```diff
@@ -0,0 +1,56 @@
+(ns ctia.task.codegen
+  ^{:doc "A task to generate swagger client libraries for a handful of languages"}
+  (:require [ctia.init :refer [start-ctia!]]
+            [clojure.java.io :as io]
+            [clojure.java.shell :as shell]
+            [ctia.properties :refer [properties]]))
+
+(def codegen-version "2.1.6")
+(def codegen-repo "http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/")
+(def supported-languages ["clojure"
+                          "java"
+                          "haskell-servant"
+                          "python"
+                          "ruby"])
+
+(def jar-uri (str codegen-repo
+                  codegen-version
+                  "/swagger-codegen-cli-"
+                  codegen-version ".jar"))
+
+(def local-jar-uri (str "/tmp/swagger-codegen-" codegen-version ".jar"))
+
+(defn spec-uri []
+  (let [port (get-in @properties [:ctia :http :port] 3000)]
+    (str "http://localhost:" port "/swagger.json")))
+
+(defn exec-command [& args]
+  (when-let [err (:err (apply shell/sh args))]
+    (println err)))
+
+(defn setup []
+  (println "starting CTIA...")
+  (start-ctia!)
```

Comment:
Might want to specify ":join? false, :silent? true" incase the defaults ever change.